### PR TITLE
feat(discover) Add http_method, http_referer and url field to transactions

### PR DESCRIPTION
**Pull request by evanh**
…tions

These fields already exist in events and errors but are missing in transactions.
This adds the columns to the table, and will be populated in a separate PR.

### CODE REVIEW

File: snuba/datasets/storages/transactions.py [hunk 1]
```diff
@@ -117,6 +117,19 @@ def transactions_migrations(
             )
         )
 
+    if "http_method" not in current_schema:
+        ret.append(
+            f"ALTER TABLE {clickhouse_table} ADD COLUMN http_method Nullable(String)"
```
Comment (JTCunning) [inline snuba/datasets/storages/transactions.py:122]: `Nullable(LowCardinality(String))`

File: snuba/migrations/snuba_migrations/transactions/0005_transactions_add_http_fields.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds the http columns populated from the Request interface that is missing from transactions.
+    """
+
+    blocking = True
```
Comment (lynnagara) [inline snuba/migrations/snuba_migrations/transactions/0005_transactions_add_http_fields.py:13]: Is there a reason it needs to be marked true? This is intended for data migrations / things that may take some time.

Comment (lynnagara) [reply snuba/migrations/snuba_migrations/transactions/0005_transactions_add_http_fields.py:13]: I guess related to this, were you planning to provide a migration that actually populates data into these columns from somewhere?

Comment (evanh) [reply snuba/migrations/snuba_migrations/transactions/0005_transactions_add_http_fields.py:13]: I'll remove it then.

File: snuba/migrations/snuba_migrations/transactions/0006_transactions_add_http_fields.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                StorageSetKey.TRANSACTIONS, "transactions_local", "http_referer"
+            ),
+        ]
+
+    def forwards_dist(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.TRANSACTIONS,
+                table_name="transactions_dist",
+                column=Column(
```
Comment (evanh) [inline snuba/migrations/snuba_migrations/transactions/0006_transactions_add_http_fields.py:66]: In other migrations this is not a materialized column, so I'll change this.

File: tests/datasets/test_transactions.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        clickhouse = (
+            get_storage(StorageKey.TRANSACTIONS)
+            .get_cluster()
+            .get_query_connection(ClickhouseClientSettings.QUERY)
+        )
+
+        event = clickhouse.execute(
+            f"SELECT url FROM transactions_local WHERE url = '{self.url}'"
+        )
+        print(event)
```
Comment (markstory) [inline tests/datasets/test_transactions.py:131]: Leftover print

File: snuba/datasets/storages/transactions.py [hunk 5]
```diff
@@ -133,6 +133,24 @@ def transactions_migrations(
             f"AFTER measurements.key"
         )
 
+    if "url" not in current_schema:
```
Comment (fpacifici) [inline snuba/datasets/storages/transactions.py:136]: Same question I had on the error table.
The url column inn the events table is a promoted tag, that means the column exists onnly for optimization and should already be accessible as `tags[url]`. 
If we are preserving that behavior there should be no need to add the column here and we could rely on the tag.

https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/events_common.py#L41-L54

**Comment by lynnagara** [comment]
I think you accidentally checked in a .DS_Store file

**Review by fpacifici — APPROVED** [review]
please address the comment before merging.

**Pull request merged**